onboarding/mlsys/1-gradients.md

```diff
@@ -0,0 +1,493 @@
+# Part 1: Calculating Gradients and Averaging Them Together
+
+Machine learning is all about getting computers to solve math problems for you.
+
+Specifically, it's about solving math problems like, "For a given mathematical
+model, what model weights would cause the model to calculate the most accurate
+predictions for a set of inputs?"
+
+In this exercise, you'll write code to solve this exact problem and for one
+particularly simple model.
+
+Afterward, you will understand basic concepts like what a gradient is and how
+gradient descent works to train models.  You will also understand how many
+features in Determined, like batch size, gradient aggregation, and
+data-parallel distributed training, really just boil down to different ways of
+averaging gradients together during gradient descent.
+
+## The Model
+
+Our model is going to be a simple linear model: `ypred = m*x` (`pred` is
+short for "predicted"). In statistics they would say there is no bias term, or
+in algebra they would say there's no constant term.  Just a straight line
+through the origin.
+
+## The Data Set
+
+Our made-up dataset has 4 data points (`true` indicates "ground truth"):
+
+- `(x=1, ytrue=1)`
+- `(x=2, ytrue=2)`
+- `(x=3, ytrue=3)`
+- `(x=4, ytrue=4)`
+
+## The Math Problem
+
+What is the best value of `m` to make our model (`ypred = m*x`) most accurately
+predict our dataset?
+
+### Q1:
+
+The answer is `m=1`.  Looking at the dataset, explain why `m=1` is correct
+before proceeding with further math.
+
+## What does "most accurately" mean, really?
+
+For this model and dataset, there is a perfect answer (`m=1`), but usually there
+isn't.  So usually instead of seeking an perfect answer, we are really seeking
+to minimize a function, called the "loss", that mathematically describes the
+goodness-of-fit between our model and the data.  There are lots of loss
```

Comment:
I like `goodness-of-fit` because I remember it from math class and it's closer to a single noun than a noun phrase (vs "how well the model fits the data").

But... actually loss would be `badness-of-fit` and that's not something I ever heard in math class...

So I rephrased this, more along the lines of what you suggested.